contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java

```diff
@@ -485,10 +480,18 @@ static void createNewCarrierAndAddVehilceTypes(Scenario scenario,
 					if (singleNewCarrier.getFixedNumberOfVehilcePerTypeAndLocation() == 0)
 						singleNewCarrier.setFixedNumberOfVehilcePerTypeAndLocation(1);
 					for (int i = 0; i < singleNewCarrier.getFixedNumberOfVehilcePerTypeAndLocation(); i++) {
-						CarrierVehicle newCarrierVehicle = CarrierVehicle.Builder.newInstance(Id.create(
+						Id<Vehicle> vehilcelId = Id.create(
 								thisType.getId().toString() + "_" + thisCarrier.getId().toString() + "_" + singleDepot
 										+ "_start" + singleNewCarrier.getVehicleStartTime() + "_" + (i + 1),
-								Vehicle.class), Id.createLinkId(singleDepot), thisType)
+								Vehicle.class);
+						while (carrierCapabilities.getCarrierVehicles().containsKey(vehilcelId)) {
+							vehilcelId = Id.create(thisType.getId().toString() + "_" + thisCarrier.getId().toString()
+									+ "_" + singleDepot + "_V" + countDepots + "_start"
+									+ singleNewCarrier.getVehicleStartTime() + "_" + (i + 1), Vehicle.class);
+							countDepots++;
+						}

```

Comment:
Since now the vehicle depots are unique links, you can revert this change (starting from line 474).

Reply:
Now I added the possibility to add a depot on an existing depot location, if no other possible link is found. Thats why I do not revert this change